express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const app = express();
+const dataJson = require('./data.json');
+const notes = Object.values(dataJson.notes);
+const fs = require('fs');
+
+const jsonMiddleware = express.json();
+app.use(jsonMiddleware);
+
+app.get('/api/notes', (req, res) => {
+  res.json(notes);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const noteId = parseInt(req.params.id, 10);
+  // eslint-disable-next-line no-console
+  console.log(notes);
+  const getId = notes.findIndex(n => n.id === noteId);
+  // eslint-disable-next-line no-console
+  console.log(getId);
+  if (noteId <= 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (dataJson.notes[noteId] === undefined) {
+    res.status(404).json({ error: `'cannot find note with id ${noteId}'` });
+  }
+  if (getId !== noteId) {
+    res.status(200).json(notes[getId]);
+  }

```

Comment:
Be sure to use `else` or `return` to guarantee that the javascript code stops when it responds. if the server responds  twice, it will throw an error and crash.